tests/unit/s2n_async_generic_cb_test.c

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "api/s2n.h"
+#include "error/s2n_errno.h"
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+#include "tls/s2n_async_generic.h"
+#include "tls/s2n_cipher_suites.h"
+#include "tls/s2n_connection.h"
+#include "tls/s2n_security_policies.h"
+#include "utils/s2n_safety.h"
+
+struct s2n_async_generic_cb_test {
+    unsigned invoke_perform : 1;
+    int result;
+
+    int invoked_count;
+    struct s2n_async_op *op;
+};
+
+int async_op_perform_in_callback(struct s2n_connection *conn, struct s2n_async_op *op, void *ctx)
```

Comment:
Is this name accurate? Won't the async op sometimes not be performed in the callback?